modules/apps/headless/headless-admin-site/headless-admin-site-impl/src/main/java/com/liferay/headless/admin/site/internal/dto/v1_0/converter/MasterPageDTOConverter.java

```
@@ -82,6 +97,60 @@ public MasterPage toDTO(
 		};
 	}
 
+	private ItemExternalReference[] _getKeywordItemExternalReferences(
+			String className, long clasSPK, long groupId)
+		throws Exception {
+
+		List<AssetTag> assetTags = _assetTagLocalService.getTags(
+			className, clasSPK);
+
+		if (ListUtil.isEmpty(assetTags)) {
+			return new ItemExternalReference[0];
+		}
+
+		return TransformUtil.unsafeTransform(
+			assetTags,
+			assetTag -> {
+				Group group = _groupLocalService.getGroup(
```

Shall we throw UnsupportedOperationException here, rather than relying on a NPE , to mark that a possible missing reference will need  to be lazily handled?

---

In this case as the assetTags are retrieved from the db and do not come form an API call I think it is safe to assume that the group exists. Please if you still think that I should throw an UnsupportedOperationException let me know